internal/fs/inode/dir.go

```diff
@@ -292,6 +292,9 @@ func (d *dirInode) lookUpChildFile(ctx context.Context, name string) (*Core, err
 
 func (d *dirInode) lookUpChildDir(ctx context.Context, name string) (*Core, error) {
 	childName := NewDirName(d.Name(), name)
+	if d.isHNSEnabled && d.bucket.BucketType() == gcs.Hierarchical {
```

Comment:
Can you create a helper method for this. i see it is used at multiple places

Reply:
Done